modules/sitemaps.php

```diff
@@ -20,6 +20,10 @@
 	exit;
 }
 
+if ( class_exists( 'Jetpack' ) && Jetpack::is_module_active( 'private' ) ) {
```

Comment:
Would it makes sense to filter the `blog_public` option via pre_get_option and utilize the existing check below?

Reply:
The problem with using `blog_public` is that parts of core expect that `blog_public` is either `1` or `0`, so adding a third value (`-1`) breaks other things.

Reply:
Linking to an earlier conversation about `blog_public` [here](https://github.com/Automattic/jetpack/pull/10333#discussion_r262339515).